tests/v2/test_peer_manager.nim

```diff
@@ -655,3 +649,52 @@ procSuite "Peer Manager":
       not pm.peerStore.peers.anyIt(it.peerId == peers[0].peerId)
       not pm.peerStore.peers.anyIt(it.peerId == peers[1].peerId)
       not pm.peerStore.peers.anyIt(it.peerId == peers[2].peerId)
+
+  asyncTest "canBeConnected() returns correct value":
+    let pm = PeerManager.new(
+      switch = SwitchBuilder.new().withRng(rng).withMplex().withNoise()
+      .withPeerStore(10)
+      .withMaxConnections(5)
+      .build(),
+      initialBackoffInSec = 1, # with InitialBackoffInSec = 1 backoffs are: 1, 2, 4, 8secs.
+      backoffFactor = 2,
+      maxFailedAttempts = 10,
```

Comment:
This would test only the happy path, can we randomize these values?
I think tests should be deterministic, but wherever math is we should be fuzzing too

Reply:
That's a good point, I am testing for failure by modifying the `maxFailedAttempts` on the line 698, but I guess some fuzzing would make sense

Reply:
This works too, or as @alrevuelta mentioned above, this is a non blocker :)